frontend/packages/console-app/src/components/tour/tour-context.ts

```diff
@@ -84,11 +91,16 @@ export const useTourValuesForContext = (): TourContextType => {
       setTourCompletionLocalStorageDataForPerspective(activePerspective, true);
     }
   };
+  const activePerspectiveRef = useRef<string>(activePerspective);
   const [tourState, tourDispatch] = useReducer<TourReducer>(tourReducer, {
     completedTour: completed,
     stepNumber: 0,
     startTour: !completed,
   });
+  if (activePerspective !== activePerspectiveRef.current) {
+    tourDispatch({ type: TourActions.initialize, payload: { completed } });
+    activePerspectiveRef.current = activePerspective;
+  }
```

Comment:
You should execute side effects inside `useEffect`
```suggestion
React.useEffect(() => {
    tourDispatch({ type: TourActions.initialize, payload: { completed } });
    setPerspective(activePerspective);
  }, [activePerspective, completed]);
```

Reply:
used `useEffect` for initializing state on perspective change